test_conformance/device_execution/device_info.cpp

```diff
@@ -99,6 +99,15 @@ int test_device_info(cl_device_id device, cl_context context, cl_command_queue q
         return -1;
     }
 
+    cl_device_type device_type;
+    err_ret = clGetDeviceInfo(device, CL_DEVICE_TYPE, sizeof(cl_device_type), &device_type, &ret_len);
```

Comment:
If we're checking for this case here do we also need to check for it in `test_create_context_from_type_device_type_all` and  `test_create_context_from_type_device_type_default`?

If we decide to keep the check here, please see the previous comments about the device type being a bitfield.